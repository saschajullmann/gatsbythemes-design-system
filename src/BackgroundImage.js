import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import { width } from 'styled-system'
import theme from './theme'

const image = props =>
  props.image ? { backgroundImage: `url(${props.image})` } : null

const height = props => (props.height ? { height: props.height } : null)

const BackgroundImage = styled.div`
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: ${props => props.theme.colors.gray};
  ${image} ${height} ${width};
`
const numberStringOrArray = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string,
  PropTypes.array
])

BackgroundImage.propTypes = {
  /** background-image url */
  image: PropTypes.string,
  width: numberStringOrArray
}

BackgroundImage.defaultProps = {
  theme: theme
}

BackgroundImage.displayName = 'BackgroundImage'

export default BackgroundImage
