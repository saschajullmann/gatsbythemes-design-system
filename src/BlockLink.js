import styled from '@emotion/styled'
import Link from './Link'

const BlockLink = styled(Link)`
  display: block;
  color: inherit;
  text-decoration: none;
`

BlockLink.displayName = 'BlockLink'

export default BlockLink
