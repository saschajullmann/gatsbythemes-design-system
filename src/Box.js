import styled from '@emotion/styled'
import { space, width, color, textAlign } from 'styled-system'
import PropTypes from 'prop-types'
import theme from './theme'

const Box = styled.div`
  ${space} ${width} ${color} ${textAlign}
`

Box.displayName = 'Box'

Box.defaultProps = {
  theme: theme
}

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
  ...textAlign.propTypes
}

export default Box
