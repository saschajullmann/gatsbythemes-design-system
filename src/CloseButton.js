import React from 'react'
import IconButton from './IconButton'
import PropTypes from 'prop-types'
import theme from './theme'

const CloseButton = props => <IconButton {...props} name="close" />

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
