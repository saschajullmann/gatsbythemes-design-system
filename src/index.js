export { default as Absolute } from './Absolute'
export { default as BackgroundImage } from './BackgroundImage'
export { default as Badge } from './Badge'
export { default as Banner } from './Banner'
export { default as BlockLink } from './BlockLink'
export { default as Box } from './Box'
export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as CloseButton } from './CloseButton'
export { default as Container } from './Container'
export { default as Divider } from './Divider'
export { default as Flag } from './Flag'
export { default as Flex } from './Flex'
export { default as GreenButton } from './GreenButton'
export { default as Heading } from './Heading'
export { default as Hide } from './Hide'
export { default as Hug } from './Hug'
export { default as Icon } from './Icon'
export { default as IconButton } from './IconButton'
export { default as IconField } from './IconField'
export { default as Image } from './Image'
export { default as Input } from './Input'
export { default as InputField } from './FormField'
export { default as FormField } from './FormField'
export { default as InputGroup } from './InputGroup'
export { default as Label } from './Label'
export { default as Link } from './Link'
export { default as OutlineButton } from './OutlineButton'
export { default as Radio } from './Radio'
export { default as RatingBadge } from './RatingBadge'
export { default as RedButton } from './RedButton'
export { default as Relative } from './Relative'
export { default as Stamp } from './Stamp'
export { default as Select } from './Select'
export { default as Text } from './Text'
export { default as Tooltip } from './Tooltip'
export { default as ToggleBadge } from './ToggleBadge'
export { default as Truncate } from './Truncate'
export { default as Checkbox } from './Checkbox'

export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export * from './theme'