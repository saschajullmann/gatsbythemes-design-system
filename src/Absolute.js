import styled from '@emotion/styled'
import Box from './Box'
import { top, right, bottom, left, zIndex } from 'styled-system'

const Absolute = styled(Box)`
  position: absolute;
  ${top} ${bottom} ${left} ${right}
  ${zIndex}
`

Absolute.propTypes = {
  ...top.propTypes,
  ...right.propTypes,
  ...bottom.propTypes,
  ...left.propTypes,
  ...zIndex.propTypes
}

Absolute.displayName = 'Absolute'

export default Absolute
