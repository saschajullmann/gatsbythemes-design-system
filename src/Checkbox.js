import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="boxChecked" size={size} data-name="checked" />
      <Icon name="boxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
    props.disabled
      ? props.theme.colors.borderGray
      : props.theme.colors.blue};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Checkbox.defaultProps = {
  size: 20
}

export default Checkbox
