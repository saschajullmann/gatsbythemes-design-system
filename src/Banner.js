import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Icon from './Icon'
import CloseButton from './CloseButton'
import Heading from './Heading'
import PropTypes from 'prop-types'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
}

const Banner = props => {
  const bannerColor = bannerColors[props.bg] || {}
  const icon = props.iconName || bannerColor.icon

  return (
    <Box
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={bannerColor.color || props.color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!icon && !!props.showIcon && (
          <Icon name={icon} mr={2} size={24} mt="-2px" />
        )}
        <Box w={1}>
          <Text textAlign={props.textAlign}>
            <Heading.h5>{props.header}</Heading.h5>
            <Text.span fontSize={1}>{props.text}</Text.span>
            {props.children}
          </Text>
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </Box>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.string,
  iconName: PropTypes.string,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string
}

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
}

export default Banner
