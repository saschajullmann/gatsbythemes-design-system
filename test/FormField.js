import React from 'react'
import PropTypes from 'prop-types'
import renderer from 'react-test-renderer'
import { renderIntoDocument } from 'react-dom/test-utils'
import { FormField, Icon, Input, Label, Select } from '../src'

afterEach(() => {
  // bust cache for propTypes
  FormField.displayName = 'FormField' + Math.random()
})

describe('FormField', () => {
  test('renders', () => {
    const json = renderer
      .create(
        <FormField>
          <Label>Email Address</Label>
          <Input id="email" />
        </FormField>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with Icon', () => {
    const json = renderer
      .create(
        <FormField>
          <Label>Email Address</Label>
          <Icon name="email" />
          <Input name="email" />
        </FormField>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with Select ', () => {
    const json = renderer
      .create(
        <FormField>
          <Select />
        </FormField>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with Select and Icon', () => {
    const json = renderer
      .create(
        <FormField>
          <Icon name="email" />
          <Select />
        </FormField>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with Label autoHide prop', () => {
    const json = renderer
      .create(
        <FormField>
          <Label autoHide>Email</Label>
          <Icon name="email" />
          <Input name="email" />
        </FormField>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('shows Label with autoHide prop and field value', () => {
    const json = renderer
      .create(
        <FormField>
          <Label autoHide>Email</Label>
          <Icon name="email" />
          <Input name="email" value="hello@example.com" />
        </FormField>
      )
      .toJSON()
    const [label] = json.children
    expect(label.props.style.opacity).toBe(1)
  })

  test('adds htmlFor prop to Label', () => {
    const json = renderer
      .create(
        <FormField>
          <Label />
          <Input name="hello" />
        </FormField>
      )
      .toJSON()
    const [label] = json.children
    expect(label.props.htmlFor).toBe('hello')
  })

  describe('propTypes', () => {
    test('warns when field is missing', () => {
      const spy = jest.spyOn(global.console, 'error')
      const err = PropTypes.checkPropTypes(
        FormField.propTypes,
        {
          children: [<Label />]
        },
        'children',
        'FormField'
      )
      expect(spy).toHaveBeenCalledTimes(1)
      spy.mockRestore()
    })

    test('warns when Label is missing', () => {
      const spy = jest.spyOn(global.console, 'error')
      PropTypes.checkPropTypes(
        FormField.propTypes,
        {
          children: [<Input name="test" />]
        },
        'children',
        'FormField'
      )
      expect(spy).toHaveBeenCalledTimes(1)
      spy.mockRestore()
    })
  })
})
