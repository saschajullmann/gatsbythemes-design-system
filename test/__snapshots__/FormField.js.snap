// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 0px;
  border-style: solid;
  border-color: #c0cad5;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  box-shadow: 0 0 0 1px #c0cad5;
}

.c2::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c2::-moz-placeholder {
  color: #4f6f8f;
}

.c2:-ms-input-placeholder {
  color: #4f6f8f;
}

.c2::placeholder {
  color: #4f6f8f;
}

.c2::-ms-clear {
  display: none;
}

.c2:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  margin: 0;
  margin-bottom: -20px;
  margin-left: 12px;
  padding-top: 6px;
  font-size: 10px;
  color: #4f6f8f;
  font-weight: 700;
}

<div>
  <label
    className="c0"
    color="gray"
    fontSize={10}
    fontWeight="bold"
    htmlFor="email"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
      }
    }
  >
    Email Address
  </label>
  <div
    className="c1"
  >
    <input
      className="c2"
      id="email"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 20,
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Icon 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 0px;
  border-style: solid;
  border-color: #c0cad5;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  box-shadow: 0 0 0 1px #c0cad5;
}

.c3::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c3::-moz-placeholder {
  color: #4f6f8f;
}

.c3:-ms-input-placeholder {
  color: #4f6f8f;
}

.c3::placeholder {
  color: #4f6f8f;
}

.c3::-ms-clear {
  display: none;
}

.c3:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  margin: 0;
  margin-bottom: -20px;
  margin-left: 40px;
  padding-top: 6px;
  font-size: 10px;
  color: #4f6f8f;
  font-weight: 700;
}

<div>
  <label
    className="c0"
    color="gray"
    fontSize={10}
    fontWeight="bold"
    htmlFor="email"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
      }
    }
  >
    Email Address
  </label>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      style={
        Object {
          "alignSelf": "center",
          "flex": "none",
          "marginLeft": 8,
          "marginRight": -32,
          "pointerEvents": "none",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
    <input
      className="c3"
      id="email"
      name="email"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 40,
          "paddingRight": undefined,
          "paddingTop": 20,
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Label autoHide prop 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 0px;
  border-style: solid;
  border-color: #c0cad5;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  box-shadow: 0 0 0 1px #c0cad5;
}

.c3::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c3::-moz-placeholder {
  color: #4f6f8f;
}

.c3:-ms-input-placeholder {
  color: #4f6f8f;
}

.c3::placeholder {
  color: #4f6f8f;
}

.c3::-ms-clear {
  display: none;
}

.c3:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  margin: 0;
  margin-bottom: -20px;
  margin-left: 40px;
  padding-top: 6px;
  font-size: 10px;
  color: #4f6f8f;
  font-weight: 700;
}

<div>
  <label
    className="c0"
    color="gray"
    fontSize={10}
    fontWeight="bold"
    htmlFor="email"
    style={
      Object {
        "height": 20,
        "opacity": 0,
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
      }
    }
  >
    Email
  </label>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      style={
        Object {
          "alignSelf": "center",
          "flex": "none",
          "marginLeft": 8,
          "marginRight": -32,
          "pointerEvents": "none",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
    <input
      className="c3"
      id="email"
      name="email"
      style={
        Object {
          "paddingBottom": 14,
          "paddingLeft": 40,
          "paddingRight": undefined,
          "paddingTop": 14,
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Select  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-left: -32px;
  color: gray;
}

.c3 {
  pointer-events: none;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #c0cad5;
  margin: 0px;
  padding-left: 12px;
  padding-right: 32px;
  padding-top: 14px;
  padding-bottom: 14px;
  font-size: 14px;
}

.c2:focus {
  outline: none;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={1}
    >
      <select
        className="c2"
        fontSize={1}
        style={
          Object {
            "paddingBottom": 14,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": 14,
            "transitionDuration": ".1s",
            "transitionProperty": "padding-top, padding-bottom",
          }
        }
      />
      <svg
        className="c3 c4"
        color="gray"
        fill="currentcolor"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`FormField renders with Select and Icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-left: -32px;
  color: gray;
}

.c4 {
  pointer-events: none;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #c0cad5;
  margin: 0px;
  padding-left: 12px;
  padding-right: 32px;
  padding-top: 14px;
  padding-bottom: 14px;
  font-size: 14px;
}

.c3:focus {
  outline: none;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

<div>
  <div
    className="c0"
  >
    <svg
      className="c1"
      fill="currentcolor"
      height={24}
      style={
        Object {
          "alignSelf": "center",
          "flex": "none",
          "marginLeft": 8,
          "marginRight": -32,
          "pointerEvents": "none",
        }
      }
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
    <div
      className="c2"
      width={1}
    >
      <select
        className="c3"
        fontSize={1}
        style={
          Object {
            "paddingBottom": 14,
            "paddingLeft": 40,
            "paddingRight": undefined,
            "paddingTop": 14,
            "transitionDuration": ".1s",
            "transitionProperty": "padding-top, padding-bottom",
          }
        }
      />
      <svg
        className="c4 c5"
        color="gray"
        fill="currentcolor"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
